ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activityConverter.service.ts

```diff
@@ -150,24 +155,49 @@ export class ActivityConverter {
                     if (!block) {
                         continue;
                     }
-                    this.buildShownField(block, inputBlock).id = inputBlock.blockGuid;
+                    this.buildShownField(block, inputBlock);
+                    block.id = inputBlock.blockGuid;
                     section.blocks.push(block);
                 } else {
                     // TODO throw exception here? For now no. Just log it as a problem
-                    this.logger.logError(this.LOG_SOURCE,
-                        `Received unknown block type with name ${inputBlock.blockType} from server`);
+                    this.logger.logError(this.LOG_SOURCE, `Received unknown block type with name ${inputBlock.blockType} from server`);
                 }
             }
         }
         return section;
     }
 
     private buildShownField<TBlock extends ActivityBlock>(block: TBlock, inputBlock: any): TBlock {
-        if (inputBlock.shown != null) {
-            block.shown = inputBlock.shown;
-        } else {
-            block.shown = true;
-        }
+        block.shown = (inputBlock.shown != null) ? inputBlock.shown : true;
         return block;
     }
+
+  private initBlockBuilders(): void {
+      this.blockBuilders = [
+          {
+              type: BlockType.Content.toUpperCase(),
+              func: (input) => this.convertContentBlock(input)
+          },
+          {
+              type: BlockType.Question.toUpperCase(),
+              func: (input) => this.questionConverter.buildQuestionBlock(input.question, input.displayNumber)
+          },
+          {
+              type: 'COMPONENT', // TODO: check - where is this type enum ?
```

Comment:
I think you can update `BlockType` enum and add this block type there.

Reply:
Good point, thanks!